_source/_posts/2022-02-25-thymeleaf-security.md

```diff
@@ -0,0 +1,670 @@
+    public Mono<Rendering> home(Authentication authentication) {
+        List<String> authorities = authentication.getAuthorities()
+                .stream()
+                .map(scope -> scope.toString())
+                .collect(Collectors.toList());
+        return Mono.just(Rendering.view("home").modelAttribute("authorities", authorities).build());
+    }
+}
+```
+
+The controller will render the `home` view and will set the authorities as a model attribute, for security checks that will be added soon.
+
+## Tweak the security configuration
+
+The default Okta starter auto-configuration will request authentication to access any page, so to customize the security, add a `SecurityConfiguration` class in the same package as before.
+
+```java
+package com.okta.developer.demo;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity;
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity;
+import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.web.server.SecurityWebFilterChain;
+import org.springframework.security.web.server.authentication.logout.RedirectServerLogoutSuccessHandler;
+import org.springframework.security.web.server.authentication.logout.ServerLogoutSuccessHandler;
+
+import java.net.URI;
+
+@EnableWebFluxSecurity
+@EnableReactiveMethodSecurity
+public class SecurityConfiguration {
+
+    @Bean
+    public ServerLogoutSuccessHandler logoutSuccessHandler(){
+        RedirectServerLogoutSuccessHandler handler = new RedirectServerLogoutSuccessHandler();
+        handler.setLogoutSuccessUrl(URI.create("/"));
+        return handler;
+    }
+
+    @Bean
+    public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
+        http
+                .authorizeExchange().pathMatchers("/").permitAll().and().anonymous()
+                .and().authorizeExchange().anyExchange().authenticated()
+                .and().oauth2Client()
+                .and().oauth2Login()
+                .and().logout().logoutSuccessHandler(logoutSuccessHandler());
+
+
+        return http.build();
+    }
+}
+```
+
+In the configuration above, access to the home `/` page is allowed to all with anonymous authentication, which is required for the authentication expression in the Thymeleaf `home.html` template to work. An authentication object must be in the security context.
+
+## Run the application
+
+Run the application with Maven:
+
+```shell
+./mvnw spring-boot:run
+```
+
+Go to http://localhost:8080 and you should see the home page and a **Sign In** button. Click the button and sign in with your Okta credentials. After signing in, you should be redirected back to the home page and see the content for authenticated users.
+
+
+{% img blog/thymeleaf-security/home-authenticated.png alt:"Home Page Authenticated" width:"800" %}{: .center-image }
+
+
```

Comment:
Remove extra line breaks around the image above.

Why do you scratch out your name in this image? It's shown in another image below.

Reply:
Because it show my toptal email, and it is not an okta email, I will blur it in the other image.